src/H5FD.c

```diff
@@ -895,56 +936,71 @@ H5FD_open(bool try, H5FD_t **_file, const char *name, unsigned flags, hid_t fapl
     if (HADDR_UNDEF == maxaddr)
         maxaddr = driver->maxaddr;
 
+    /* clang-format off */
     /* Try dispatching to file driver */
     if (try) {
         H5E_PAUSE_ERRORS
         {
-            file = (driver->open)(name, flags, fapl_id, maxaddr);
+            /* Prepare & restore library for user callback */
+            H5_BEFORE_USER_CB(FAIL)
+            {
+                file = (driver->open)(name, flags, fapl_id, maxaddr);
+            }
+            H5_AFTER_USER_CB(FAIL)
         }
         H5E_RESUME_ERRORS
 
         /* Check if file was not opened */
         if (NULL == file)
             HGOTO_DONE(SUCCEED);
     }
-    else if (NULL == (file = (driver->open)(name, flags, fapl_id, maxaddr)))
-        HGOTO_ERROR(H5E_VFL, H5E_CANTOPENFILE, FAIL, "can't open file");
-
-    /* Set the file access flags */
-    file->access_flags = flags;
-
-    /* Fill in public fields. We must increment the reference count on the
-     * driver ID to prevent it from being freed while this file is open.
-     */
-    file->driver_id = driver_prop.driver_id;
-    if (H5I_inc_ref(file->driver_id, false) < 0)
-        HGOTO_ERROR(H5E_VFL, H5E_CANTINC, FAIL, "unable to increment ref count on VFL driver");
-    file->cls     = driver;
-    file->maxaddr = maxaddr;
-    if (H5P_get(plist, H5F_ACS_ALIGN_THRHD_NAME, &file->threshold) < 0)
-        HGOTO_ERROR(H5E_PLIST, H5E_CANTGET, FAIL, "can't get alignment threshold");
-    if (H5P_get(plist, H5F_ACS_ALIGN_NAME, &file->alignment) < 0)
-        HGOTO_ERROR(H5E_PLIST, H5E_CANTGET, FAIL, "can't get alignment");
-
-    /* Retrieve the VFL driver feature flags */
-    if (H5FD__query(file, &file->feature_flags) < 0)
-        HGOTO_ERROR(H5E_VFL, H5E_CANTINIT, FAIL, "unable to query file driver");
-
-    /* Increment the global serial number & assign it to this H5FD_t object */
-    if (++H5FD_file_serial_no_g == 0) {
-        /* (Just error out if we wrap around for now...) */
-        HGOTO_ERROR(H5E_VFL, H5E_CANTINIT, FAIL, "unable to get file serial number");
-    } /* end if */
-    file->fileno = H5FD_file_serial_no_g;
-
-    /* Start with base address set to 0 */
-    /* (This will be changed later, when the superblock is located) */
-    file->base_addr = 0;
+    else {
+        /* Prepare & restore library for user callback */
+        H5_BEFORE_USER_CB(FAIL)
+        {
+            file = (driver->open)(name, flags, fapl_id, maxaddr);
+        }
+        H5_AFTER_USER_CB(FAIL)
+        if (NULL == file)
+            HGOTO_ERROR(H5E_VFL, H5E_CANTOPENFILE, FAIL, "can't open file");
+    }
+/* clang-format on */
```

Comment:
Did the formatter get confused here and remove the indent?

Reply:
Yeah, it was getting confused and I got tired of messing around with it.  :-/